API/SearchService/__init__.py

```diff
@@ -16,10 +16,15 @@ def main(req: func.HttpRequest,
         keyword = get_param_str(req, "keyword", required=True)
         search_type = get_param_search_type(req, "type", default=SearchType.All)
         top_num = get_param_int(req, "top_num", default=5)
+        if top_num <=0 or top_num > 20:
+            raise ParameterException("Illegal parameter: the parameter 'top_num' must be in the range 1-20")
+        match_type = get_param_match_type(req, "match_type", default=MatchType.All)
     except ParameterException as e:
         return func.HttpResponse(e.msg, status_code=400)
-    results = get_search_results(
-        build_search_statement(keyword), top_num, search_type.get_search_fields())
+    results = get_search_results(build_search_statement(keyword, match_type), top_num, search_type.get_search_fields())
+    if len(keyword.split()) > 1 and len(results) < top_num and match_type == MatchType.All:
+        or_results = get_search_results(build_or_search_statement(keyword), top_num, search_type.get_search_fields())
```

Comment:
```suggestion
        or_results = get_search_results(build_or_search_statement(keyword), top_num-len(results), search_type.get_search_fields())
```

Reply:
There can be some duplicate search results in `and` search and `or` search. So I don't think it's appropriate to limit extra search `top_num`.

Reply:
Got it, thanks~

Reply:
However, do we have logic to limit the number of returned results according to `top_num`? 
If not, is it possible that the number we return will be greater than `top_num`?

Reply:
Sorry. I forget to put a limit on merged results.

Reply:
We should add 
```
results = results[:min(top_num, len(results))]
```
after next line `append_results(results, or_results)`.